### PR TITLE
Configure services for SlugGenerator & ExcerptGenerator

### PR DESCRIPTION
**Pull request by julietdaniel**
Register the SlugGenerator and ExcerptGenerator as actual services in Startup.cs's Configure Service method, and then consume them from the DI container wherever they're used.

### CODE REVIEW

File: BlogTemplate/Models/BlogDataStore.cs [hunk 1]
```diff
@@ -115,17 +115,17 @@ public Comment FindComment(Guid UniqueId, Post post)
...
-            XElement tagsNode = new XElement("Tags");
-            foreach (string tag in post.Tags)
-            {
-                tagsNode.Add(new XElement("Tag", tag));
-            }
-            rootNode.Add(tagsNode);
-
-            return rootNode;
-        }
+        //public XElement AddTags(Post post, XElement rootNode)
```
Comment (Eilon) [inline BlogTemplate/Models/BlogDataStore.cs]: If this code is no longer needed, I recommend removing it.

Comment (Eilon) [inline BlogTemplate/Models/BlogDataStore.cs]: (Including other commented-out code.)

File: BlogTemplate/Pages/Edit.cshtml.cs [hunk 2]
```diff
@@ -15,10 +15,14 @@ namespace BlogTemplate.Pages
     public class EditModel : PageModel
     {
         private BlogDataStore _dataStore;
+        private SlugGenerator _slugGenerator;
```
Comment (Eilon) [inline BlogTemplate/Pages/Edit.cshtml.cs]: Should mark these fields as `readonly` as a best practice. It ensures that the fields can be set only from the ctor.

File: BlogTemplate/Pages/Edit.cshtml.cs [hunk 3]
```diff
@@ -64,16 +68,15 @@ private void UpdatePost(Post newPost, string slug)
         {
             oldPost = _dataStore.GetPost(slug);
             newPost.PubDate = oldPost.PubDate;
+            //newPost.Tags = Request.Form["Tags"][0].Replace(" ", "").Split(",").ToList();
```
Comment (Eilon) [inline BlogTemplate/Pages/Edit.cshtml.cs]: More commented lines 😄 Remove? Or keep?

File: BlogTemplate/Pages/New.cshtml.cs [hunk 4]
```diff
@@ -18,10 +16,14 @@ public class NewModel : PageModel
     {
         const string StorageFolder = "BlogFiles";
         private BlogDataStore _dataStore;
+        private SlugGenerator _slugGenerator;
```
Comment (Eilon) [inline BlogTemplate/Pages/New.cshtml.cs]: Also mark `readonly`?

File: BlogTemplate/Pages/New.cshtml.cs [hunk 5]
```diff
@@ -53,13 +55,13 @@ public IActionResult OnPostSaveDraft()
 
         private void SavePost(Post post)
         {
-            SlugGenerator slugGenerator = new SlugGenerator(_dataStore);
-            Post.Slug = slugGenerator.CreateSlug(Post.Title);
+            //Post.Tags = Request.Form["Tags"][0].Replace(" ", "").Split(",").ToList();
```
Comment (Eilon) [inline BlogTemplate/Pages/New.cshtml.cs]: Commented code?

File: BlogTemplate/Pages/Edit.cshtml.cs (hunk 3 above)
Comment (julietdaniel) [inline BlogTemplate/Pages/Edit.cshtml.cs]: Whoops, don't know why these commented-out lines keep showing up! I merged in the most recent update of master, which had deleted the commented-out lines of code. Guess I'll need to find a different way to take care of these...

**Pull request merged**